internal/testing/factory/factory.go

```diff
@@ -341,15 +342,12 @@ func (factory *Factory) MakeModel(c *gc.C, params *ModelParams) *state.State {
 		cfgType = "kubernetes"
 	}
 
-	var modelUUID uuid.UUID
-	if params.UUID != nil {
-		modelUUID = *params.UUID
-	} else {
-		modelUUID = uuid.MustNewUUID()
+	if params.UUID == coremodel.UUID("") {
```

Comment:
You should find the tests that are using the testing factory and remove the usages of them - either with new tests, or with skipping them via TestStub method.